src/server/multi_svr.c

```diff
@@ -756,31 +852,47 @@ init_node_from_bstat(struct batch_status *bstat, server_t *psvr)
 	pbs_node *pnode;
 	pbs_list_head attrs;
 	struct batch_status *cur;
+	int rc = 0;
 
 	for (cur = bstat; cur; cur = cur->next) {
+
 		pnode = calloc(1, sizeof(pbs_node));
+		if (!pnode) {
+			log_errf(PBSE_SYSTEM, __func__, "Failed to allocate memory for pnode");
+			rc = -1;
+			continue;
```

Comment:
Don't we want to return immediately in case of memory failure ?

Reply:
trying to cache the remaining elements if the memory is available for future operations.